src/main/java/frc/robot/scoreStates/ShootState.java

```diff
@@ -44,20 +60,49 @@ public Command driveToClosestOptimalPoint() {
                     return AutoBuilder.pathfindToPose(
                             new Pose2d(
                                     optimalTranslation,
-                                    Utils.calculateOptimalRotation(
+                                    Utils.calcRotationToTranslation(
                                             optimalTranslation, speakerPose)),
                             Constants.AUTO_CONSTRAINTS);
                 },
                 Set.of(SwerveDrive.getInstance()));
     }
 
     @Override
-    public Command stateInitialize() {
-        return null;
+    public Command initializeCommand() {
+        return Commands.none();
+    }
+
+    @Override
+    public Command initializeSubsystem() {
+        return Commands.parallel(
```

Comment:
1. Didn't we say that the shooter and hood warmups will be default commands? I'm not saying the current implementation is bad, just want to make sure.
2. This code doesn't handle the "don't warmup if we're far away from the Speaker".
3. Reposting the second part of that reply: https://github.com/Galaxia5987/robot-2024/pull/26#discussion_r1480053705

Reply:
We changed how the shoot state works, Emma is working on it on a different branch which will be merged to here once she is done.